overdue/src/test/java/org/killbill/billing/overdue/config/TestOverdueConfig.java

```diff
@@ -75,4 +80,33 @@ public void testParseConfig() throws Exception {
 
         Assert.assertEquals(c.getOverdueStatesAccount().getFirstState().getName(), "OD1");
     }
+
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testBadOverdueXml() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("BadOverdue.xml");
```

Comment:
I would suggest renaming the test to `testTotalUnpaidInvoiceBalanceEqualsOrExceedsCondition`. Also, I think it would be better to create a different class for this test, but I will let @sbrossie confirm

Reply:
> I would suggest renaming the test to testTotalUnpaidInvoiceBalanceEqualsOrExceedsCondition
Agreed.

And also change the name of the test, because after your fix this is **not** a 'bad' overdue XML config ...

> Also, I think it would be better to create a different class for this test, but I will let @sbrossie confirm

We could go either way, since we already have a Test class for checking xml config, adding one test inside is fine with me.